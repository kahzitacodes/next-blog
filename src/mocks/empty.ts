import { EmptyProps } from '@/components/empty'

export const emptyMock: EmptyProps = {
  title: 'Oops!',
  description: 'Page not found'
}
