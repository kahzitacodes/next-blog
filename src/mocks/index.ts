export * from './navigation'
export * from './profile'
export * from './social-links'
export * from './post'
export * from './pre'
export * from './pagination'
