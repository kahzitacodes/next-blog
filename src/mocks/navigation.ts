import { NavItem } from '@/types'

export const navigationMock: NavItem[] = [
  {
    title: 'Home',
    href: '/'
  },
  {
    title: 'About',
    href: '/about'
  },
  {
    title: 'Contact',
    href: '/contact'
  }
]
