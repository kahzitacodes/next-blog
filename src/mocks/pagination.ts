export const paginationMock = {
  currentPage: 1,
  totalPages: 10
}
