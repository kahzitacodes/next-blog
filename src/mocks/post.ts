import { Post } from '@/types'

export const postMock: Post = {
  $frontMatter: {
    date: '2025-04-04',
    description: 'Description',
    image: '/assets/images/image.png',
    title: 'Post title',
    tags: ['react']
  },
  $readingTime: 3,
  body: {
    code: '',
    raw: ''
  },
  id: '1',
  slug: 'post-3'
}
