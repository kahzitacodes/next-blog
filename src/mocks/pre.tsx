import { PreProps } from '@/components/pre'

export const preMock: PreProps = {
  children: <h1>Title</h1>
}
