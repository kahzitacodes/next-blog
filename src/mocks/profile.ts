import type { ProfileProps } from '@/components'

export const profileMock: ProfileProps = {
  avatar: '/assets/images/avatar.jpeg',
  name: 'Karine Pereira',
  description: 'Lorem ipsum dolor sit amet.'
}
