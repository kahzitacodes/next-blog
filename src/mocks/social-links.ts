import { NavItem } from '@/types'

export const socialLinksMock: NavItem[] = [
  {
    title: 'Instagram',
    href: 'http://instagram.com',
    icon: 'Instagram'
  },
  {
    title: 'Twitter',
    href: 'http://twitter.com',
    icon: 'Twitter'
  },
  {
    title: 'LinkedIn',
    href: 'http://linkedin.com',
    icon: 'Linkedin'
  },
  {
    title: 'GitHub',
    href: 'http://github.com',
    icon: 'Github'
  }
]
