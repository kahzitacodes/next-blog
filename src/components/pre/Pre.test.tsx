import { render, screen } from '@testing-library/react'
import { Pre } from '.'
import { preMock } from '@/mocks'

describe('<Pre />', () => {
  it('should render successfully', () => {
    const { container } = render(<Pre dataTestid="pre" {...preMock} />)

    expect(screen.getByTestId('pre')).toBeInTheDocument()
    expect(screen.getByRole('heading', { name: /Title/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
