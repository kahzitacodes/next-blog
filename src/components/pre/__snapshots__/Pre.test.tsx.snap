// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pre /> should render successfully 1`] = `
<div
  class="text-md w-full my-6 overflow-x-auto rounded-lg bg-zinc-900"
  data-testid="pre"
>
  <div
    class="flex w-full justify-between rounded-t-lg border-zinc-950 border-b text-zinc-500 px-6 py-3"
  >
    shel
  </div>
  <pre
    class="p-4 overflow-x-auto text-base [&>code]:bg-transparent bg-zinc-800"
  >
    <h1>
      Title
    </h1>
  </pre>
</div>
`;
